rrtmgp/kernels-openacc/mo_gas_optics_kernels.F90

```diff
@@ -599,10 +672,12 @@ subroutine compute_Planck_source(                        &
     ! Map to g-points
     !
     !$acc parallel loop collapse(3)
-    do igpt = 1, ngpt
+    do icol = 1, ncol, 2
       do ilay = 1, nlay
-        do icol = 1, ncol
-          lay_src(igpt,ilay,icol) = pfrac(igpt,ilay,icol) * planck_function(gpoint_bands(igpt),ilay,icol)
+    do igpt = 1, ngpt
+          lay_src(igpt,ilay,icol  ) = pfrac(igpt,ilay,icol  ) * planck_function(gpoint_bands(igpt),ilay,icol)
+          if (icol <= ncol) &
```

Comment:
Shouldn't this test (line 679) be `icol < col`? But why is the test even needed?

Reply:
Whoops. Indeed it has to be `icol < ncol`. That is because I unrolled the loop, and `icol+1` may exceed the array bounds if `ncol` is odd